### PR TITLE
feat(data-table-filter): apply selection of operator to data-table-filter

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci`, minor refactoring, etc.)
- [ ] Need discussion
- [ ] Not that difficult
- [ ] Approved feature branch merge to master

### Description
- Add query filter operator selection

- ASIS: `apiFilter` -> `key:value filter`(`in` query)  -> `consoleFilter` -> `apiFilter`(request)
- TOBE: `apiFilter` -> `customApiFilter` -> `apiFilter`(request)

image

### CODE REVIEW

File: apps/web/src/common/modules/widgets/types/widget-model.ts [hunk 1]
```diff
@@ -27,9 +29,11 @@ export interface DataTableAddOptions {
     date_format?: DateFormat;
     additional_labels?: AdditionalLabels;
     timediff?: TimeDiff;
-    filter?: any[];
-    filter_or?: any[];
+    filter?: DataTableQueryFilter[];
+    filter_or?: DataTableQueryFilter[];
 }
+export type DataTableQueryFilter = ApiFilter;
```
Comment (yuda110) [inline apps/web/src/common/modules/widgets/types/widget-model.ts:35]: praise: good!!!

File: apps/web/src/common/modules/widgets/types/widget-data-table-type.ts [hunk 2]
```diff
@@ -45,3 +47,5 @@ export interface TransformDataTableInfo {
 export type DataTableAlertModalMode = 'DELETE'|'DELETE_UNABLED'|'RESET';
 
 export type JoinRestrictedMap = Record<string, boolean>; // { {id}: true }
+
+export type DataTableQueryFilterForDropdown = Omit<DataTableQueryFilter, 'v'> & { v: MenuItem[] };
```
Comment (yuda110) [inline apps/web/src/common/modules/widgets/types/widget-data-table-type.ts:51]: praise: nice usage

File: apps/web/src/common/modules/widgets/_components/WidgetFormDataTableCardFilters.vue [hunk 3]
```diff
@@ -309,40 +326,16 @@ onMounted(() => {
...
-                                   multi-selectable
-                                   appearance-type="stack"
-                                   show-select-marker
-                                   :init-selected-with-handler="!!GROUP_BY_TO_VAR_MODELS[item.name] || props.sourceType === DATA_SOURCE_DOMAIN.ASSET"
-                                   :show-delete-all-button="false"
-                                   :page-size="10"
-                                   @update:selected="handleUpdateFilterDropdown(item.name, $event)"
-                />
-            </div>
+            <widget-form-data-table-card-f-ilters-item v-for="(item) in state.selectedItems"
```
Comment (yuda110) [inline apps/web/src/common/modules/widgets/_components/WidgetFormDataTableCardFilters.vue:329]: quibble: f-ilters ??

Comment (piggggggggy) [reply apps/web/src/common/modules/widgets/_components/WidgetFormDataTableCardFilters.vue:329]: Holy shit..

File: apps/web/src/common/modules/widgets/_components/WidgetFormDataTableCardFilters.vue [hunk 4]
```diff
@@ -298,7 +312,10 @@ onMounted(() => {
             ...state.selectedItems,
             { name: filter.k, label: selectedFilteritemLabel },
         ];
-        state.selectedItemsMap[filter.k] = filter.v.map((d) => ({ name: d }));
+        state.selectedItemsMap[filter.k] = {
+            ...filter,
+            v: filter.v.map((d) => ({ name: d })),
+        };
```
Comment (yuda110) [inline apps/web/src/common/modules/widgets/_components/WidgetFormDataTableCardFilters.vue:318]: nitpick: here!

**Review by yuda110 — APPROVED** [review]
LGTM!!

**Pull request merged**